offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+    protected static final DecimalFormat PERCENTAGE_FORMATTER = new DecimalFormat("#%")
+
+    /**
+     * Creates a new OfferTemplate using the provided document as template document.
+     * <p>Please note: It is important that the template document provides the following css selectors</p>
+     * <ul>
+     *  <li>.currency-symbol</li>
+     *  <li>.vat-percentage</li>
+     *  <li>.overheads-percentage</li>
+     *  <li>#customer-name</li>
+     *  <li>#customer-organisation</li>
+     *  <li>#customer-street</li>
+     *  <li>#customer-postal-code</li>
+     *  <li>#customer-city</li>
+     *  <li>#customer-country</li>
+     *  <li>#qbic-ad</li>
+     *  <li>#project-manager-name</li>
+     *  <li>#project-manager-street</li>
+     *  <li>#project-manager-city</li>
+     *  <li>#project-manager-email</li>
+     *  <li>#doctype-desc</li>
+     *  <li>#offer-date</li>
+     *  <li>#offer-identifier</li>
+     *  <li>#offer-expiry-date</li>
+     *  <li>#project-title</li>
+     *  <li>#project-description</li>
+     *  <li>#experimental-design</li>
+     *  <li>.overheads  .data-generation  > .costs</li>
+     *  <li>.overheads  .data-analysis  > .costs</li>
+     *  <li>.overheads  .data-management > .costs</li>
+     *  <li>.overheads  > .totals-section  > .costs</li>
+     *  <li>.total-costs .net > .costs</li>
+     *  <li>.total-costs .discounts > .costs</li>
+     *  <li>.total-costs .overheads > .costs</li>
+     *  <li>.total-costs .vat > .costs</li>
+     *  <li>.total-costs > .totals-section > .costs</li>
+     *  <li>#data-generation-items</li>
+     *  <li>#data-generation-items > tfoot .costs</li>
+     *  <li>#data-analysis-items</li>
+     *  <li>#data-analysis-items > tfoot .costs</li>
+     *  <li>#data-management-items</li>
+     *  <li>#data-management-items > tfoot .costs</li>
+     * </ul>
+     * @param document the template document
+     * @since 1.0.0
+     */
+    OfferTemplate(Document document) {
+        this.document = document
+    }
+
+    /**
+     * Fills the template with the specified content
+     * @param content the content to be parsed and replaced in the offer
+     * @return a document filled with the provided content based on this template
+     * @since 1.1.0
+     */
+    Document fill(OfferContent content) {
+        // make a copy of the template
+        Document result = this.document.clone()
+        // fill customer information
+        fillCustomerInformation(result, content)
+        // fill project manager information
+        fillProjectManagerInformation(result, content)
+        // fill offer information
+        fillOfferInformation(result, content)
+        // fill cost summaries
+        fillCostSummary(result, content)
+        // fill tables table
+        fillDataGenerationItems(result, content)
+        fillDataAnalysisItems(result, content)
+        fillDataManagementItems(result, content)
+        // fill general information
+        fillGeneralInformation(result, content)
```

Comment:
I still think the name is not a good choice, since the method does not fill "general information". General information in the offer context would be for me: Quotation details, ... stuff from the first page or at least some information or text at all. There are only set some symbols here, so please adjust the naming to that :)